runtime/server/x86_gpu/scripts/convert.py

```diff
@@ -0,0 +1,100 @@
+    parser.add_argument('--config', required=True, help='config file')
+    parser.add_argument('--vocab', required=True, help='vocabulary file, words.txt')
+    parser.add_argument('--model_repo', required=True, help='model repo directory')
+    parser.add_argument('--onnx_model_dir', default=True, type=str, required=False,
+                        help="onnx model path")
+    parser.add_argument('--lm_path', default=None, type=str, required=False,
+                        help="the additional language model path")
+    args = parser.parse_args()
+    with open(args.config, 'r') as fin:
+        configs = yaml.load(fin, Loader=yaml.FullLoader)
+
+    with open(os.path.join(args.onnx_model_dir, 'config.yaml'), 'r') as fin:
+        onnx_configs = yaml.load(fin, Loader=yaml.FullLoader)
+
+    model_params = {
+        "#beam_size": 10,
+        "#num_mel_bins": 80,
+        "#frame_shift": 10,
+        "#frame_length": 25,
+        "#sample_rate": 16000,
+        "#encoder_out_feat_size": 256,
+        "#lm_path": "",
+        "#bidecoder": 0,
+        "#vocabulary_path": "",
+        "#vocab_size": 0,
+        "#DTYPE": "FP32"
+    }
+
+
+    # fill values
+    model_params["#beam_size"] = onnx_configs["beam_size"]
+    if onnx_configs["fp16"]:
+        model_params["#DTYPE"] = "FP16"
+    feature_conf = configs["dataset_conf"]["fbank_conf"]
+    model_params["#num_mel_bins"] = feature_conf["num_mel_bins"]
+    model_params["#frame_shift"] = feature_conf["frame_shift"]
+    model_params["#frame_length"] = feature_conf["frame_length"]
+    dataset_conf = configs["dataset_conf"]["resample_conf"]
+    model_params["#sample_rate"] = dataset_conf["resample_rate"]
+    model_params["#encoder_out_feat_size"] = configs["encoder_conf"]["output_size"]
+    model_params["#lm_path"] = args.lm_path
+    if configs["decoder"].startswith("bi"):
+        model_params["#bidecoder"] = 1
+    model_params["#vocabulary_path"] = args.vocab
+    # read vocabulary
+    num_vocab = len(open(args.vocab, "r", encoding="utf-8").readlines())
+    model_params["#vocab_size"] = num_vocab
+
+    for model in os.listdir(args.model_repo):
+        template = "config_template.pbtxt"
+        if "decoder" == model and model_params["#bidecoder"] == 0:
+            template = "config_template2.pbtxt"
+
+        model_dir = os.path.join(args.model_repo, model)
+        out = os.path.join(model_dir, "config.pbtxt")
+        out = open(out, "w")
+
+        with open(os.path.join(model_dir, template), "r", encoding="utf-8") as f:
+            for line in f:
+                if line.startswith("#"):
+                    continue
+                for key, value in model_params.items():
+                    if key == "#encoder_out_feat_size" and \
+                       configs["encoder_conf"].get("causal", False) and \
+                       model == "encoder":
+                        value = "-1"
+                    line = line.replace(key, str(value))
```

Comment:
plz add some comments for this special if-branch, why encoder_out_feat_size should be -1 if encoder uses causal conv? If so , we will get shape [-1, -1] in `model_repo/encoder/config_template.pbtxt::line38`, is it possible for onnx or triton to handle such a strange shape?

Reply:
It seems very strange that the exported model including those transformers and offline conformers all have a -1 (dynamic) output size. We need to take a close look at this issue. But it doesn't affect deployment.

Reply:
i c, i think we'd better add some notes for this if-branch to let users know the reason for doing this and leave the code as it is. could u add some comments like [this](https://github.com/wenet-e2e/wenet/pull/786/files#diff-c1fac5c7b9cb061736a33b160029d301823b2d52e2d9f1c74f733d2de54d7d8fR135-R136)? Thx.

Reply:
Sure.